internal/cortex/querier/queryrange/queryrange.proto

```diff
@@ -66,13 +66,19 @@ message PrometheusInstantQueryData {
 
 message PrometheusInstantQueryResult {
   oneof result {
-    cortexpb.Sample sample = 1;
-    Vector samples = 2;
+    cortexpb.Sample scalar = 1;
+    StringSample stringSample = 2;
+    Vector vector = 3;
+    Matrix matrix = 4;
```

Comment:
Handled all 4 cases mentioned in https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries.